### PR TITLE
Boost: Fix PHP linting setup and issues

### PR DESCRIPTION
**Pull request by davidlonjon**
This PR does fix PHP lingin setup and issues.

#### Changes proposed in this Pull Request:

* Make sure the Boost composer related linting commands are using the linting command from the composer file at the base of the repo so it can find the correct coding standards. Few things to note:
    - The drawback with this method though is that the linting cannot be done on a specific file from the Boost composer/pnpm  linting commands. Running `composer lint` (or `pnpm run php:lint`) or `composer fix` (or `pnpm run php:fix`) from the Jetpack Boost folder will run linting for all the PHP files.
    -  An alternate solution could have been to add the `automattic/jetpack-codesniffer` and `dealerdirect/phpcodesniffer-composer-installer` dev packages in Boost composer file but I thought we might want to avoid it so we just refer to the PHP linting commands from the composer file at the top of the repo.
    - Another alternate solution would have been to actually remove any PHP linting process from the Boost plugin and rely only on the PHP linting commands from the root of the repo such as using the commands `pnpm run php:lint projects/plugins/boost/` or `pnpm run php:fix projects/plugins/boost/`. With this approach a single file can be lint such as `pnpm run php:lint projects/plugins/boost/jetpack-boost.php`. In any case this solution is already available even with the solution in this PR.
- Rename the PHP linting commands in the Boost `package.json` file to be streamlined with the ones from the `package.json` file at the root of the repo
- Add an PHPCS config to ignore WordPress included in plugins.
- Fixes a large bunch of PHPCS and trying to make a few things more consistent across different files.

#### Jetpack product discussion

N/A

#### Does this pull request change what data or activity we track or use?

N/A

#### Testing instructions:

* In the Boost plugins folder you can run PHPCS linting by running the command `pnpm run php:lint` (or `pnpm run php:fix` for fixing issues). The linting should complete successfully.
* alternatively at the base of the monorepo you can run PHPCS linting by running the command `pnpm run php:lint projects/plugins/boost/` (or `pnpm run php:fix projects/plugins/boost/` for fixing issues). The linting should complete successfully.

### CODE REVIEW

File: projects/plugins/boost/composer.json [hunk 1]
```diff
@@ -30,11 +30,11 @@
...
 	"scripts": {
-		"format": [
-			"phpcbf ."
+		"fix": [
+			"composer --working-dir=../../../ phpcs:fix ./projects/plugins/boost"
 		],
 		"lint": [
-			"phpcs ."
+			"composer --working-dir=../../../ phpcs:lint ./projects/plugins/boost"
 		],
```
Comment (jeherve) [inline projects/plugins/boost/composer.json:38]: Do you think that is still needed now that we have the tools at the monorepo level?

**Review by jeherve — APPROVED** [review]
Nice cleanup work. I'll approve and I would encourage you to merge it quick in order to avoid having to do any rebases on this. :) I only have one question, but that can be addressed in a follow-up PR.

**Review by jeherve — APPROVED** [review]
Re-approving, I had missed the failed check for the lock file. 👍

File: projects/plugins/boost/composer.json (hunk 1 above)
Comment (davidlonjon) [reply projects/plugins/boost/composer.json:38]: Yes I was debating it. I think I will to discuss with @thingalon about the commands we have in Boost composer.json and package.json files. I will definitely address it in another PR.

**Comment by davidlonjon** [comment]
@thingalon as @jeherve has given the 👍, I'll merge it already so I can build on it and port the left-over PRs from the legacy repo to the code base.

**Pull request merged**